haystack/preview/components/file_converters/markdown.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+from tqdm import tqdm
+
+from haystack.preview import Document, component
+from haystack.preview.dataclasses import ByteStream
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport("Run 'pip install markdown-it-py mdit_plain'") as markdown_conversion_imports:
+    from markdown_it import MarkdownIt
+    from mdit_plain.renderer import RendererPlain
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class MarkdownToDocument:
+    """
+    Converts a Markdown file into a text Document.
+
+    Usage example:
+    ```python
+    from haystack.preview.components.file_converters.markdown import MarkdownToDocument
+
+    converter = MarkdownToDocument()
+    results = converter.run(sources=["sample.md"])
+    documents = results["documents"]
+    print(documents[0].text)
+    # 'This is a text from the markdown file.'
+    ```
+    """
+
+    def __init__(self, table_to_single_line: bool = False, progress_bar: bool = True):
+        """
+        :param table_to_single_line: Convert contents of the table into a single line. Defaults to False.
+        :param progress_bar: Show a progress bar for the conversion. Defaults to True.
+        """
+        markdown_conversion_imports.check()
+
+        self.table_to_single_line = table_to_single_line
+        self.progress_bar = progress_bar
+
+    @component.output_types(documents=List[Document])
+    def run(self, sources: List[Union[str, Path, ByteStream]], metadata: Optional[List[Dict[str, Any]]] = None):
+        """
+        Reads text from a markdown file and executes optional preprocessing steps.
+
+        :param file_path: Path of the file to convert.
+        :param metadata: Optional list of metadata to attach to the Documents.
+        The length of the list must match the number of paths. Defaults to `None`.
+        """
+        if self.table_to_single_line:
+            parser = MarkdownIt(renderer_cls=RendererPlain).enable("table")
+        else:
+            parser = MarkdownIt(renderer_cls=RendererPlain)
```

Comment:
```suggestion
    parser = MarkdownIt(renderer_cls=RendererPlain)
        if self.table_to_single_line:
            parser.enable("table")
```
(I haven't checked if it works properly)

Reply:
I did a quick check, this works fine. I will make this change.